### PR TITLE
RSDK-2240 Remove Temp Endpoints from SLAM

### PR DESCRIPTION
**Pull request by jeremyrhyde**
This PR removes the last links to the temporary endpoints created during Map Representation. These include:
- GetPositionNew
- GetPointCloudMapStream
- GetInternalStateStream

The only place these still existed was in the `server.go` file and its associated tests, although a few comments were also updated in the `client_test.go`.

Predominantly, the only places stream still exists are in the client/server tests which are comparing testing unary and streaming calls and the comments in `slam.go`/`grpchelper.go` associated with the concatenation functions.

Examples: 
```
// Helper function that concatenates the chunks from a streamed grpc endpoint.
```
```
// GetInternalStateCallback helps a client request the internal state stream from a SLAM server,
// returning a callback function for accessing the stream data.
```

If you believe these should change leave a comment below

JIRA Ticket: https://viam.atlassian.net/browse/RSDK-2240

### CODE REVIEW

File: services/slam/client_test.go
Comment (jeremyrhyde) [inline services/slam/client_test.go:118]: Optional removal, I'm in favor to keep pattern with server tests

Comment (JohnN193) [reply services/slam/client_test.go:118]: this is fine, we have a [ticket](https://viam.atlassian.net/browse/RSDK-2130?atlOrigin=eyJpIjoiOTg4M2NhYjA2OGU0NDFkOGEzNTQ4YTFhZGJiNDBmOWYiLCJwIjoiaiJ9) to make server tests into individual t.runs, which should probably include these tests too

**Review by JohnN193 — APPROVED** [review]
LGTM, but to be clear these changes are blocked by the app bump right?

File: services/slam/client_test.go
Comment (nicksanford) [reply services/slam/client_test.go:118]: I agree with john. Lets address client tests not being in t.Runs in the ticket he linked.

**Review by nicksanford — APPROVED** [review]
Approved, assuming it is not merged until https://github.com/viamrobotics/app/pull/1358 is on production

File: services/slam/client_test.go
Comment (EshaMaharishi) [reply services/slam/client_test.go:118]: [note to self] we took out the word "stream" everywhere that we aren't specifically testing the streaming aspect of the api

**Review by EshaMaharishi — APPROVED** [review]

**Pull request merged**
